base/loading.jl

```diff
@@ -4148,6 +4177,17 @@ macro __DIR__()
     return isempty(_dirname) ? pwd() : abspath(_dirname)
 end
 
+is_core_or_base(mod::Module) = mod === Core || mod === Base
+function prepare_compiler_stub_image!()
+    ccall(:jl_add_to_module_init_list, Cvoid, (Any,), Compiler)
+    # Drop any dependencies on loaded modules. We know Base.Compiler only depends
+    # on very basic `Base` functionality, so we don't want to pick up dependencies
+    # on other stdlibs that may be loaded.
```

Comment:
This seems like could lead to some significant soundness issues, as we rely on this list being correct and especially on it being complete. Please don't do this unsafe mutation of it to lose entries.

Reply:
Yes, I can drop this change. Originally, the compiler was supposed to be a completely independent module, but now REPL uses is, so it does need to be reachable.